docs/Manual/06 Plans - Caching.md

```diff
@@ -45,24 +45,27 @@ AA = plan.cache(A, level=2, layout=acc.Array.Layout.FIRST_MAJOR)
+Note that the current active blocks of an array are nested, and their sizes are monotonic (nondecreasing) in their level. Therefore, we can also select the largest active block that does not exceed a certain number of elements. 
 ```python
 AA = plan.cache(A, max_elements=1024)
 ```
 
 
 ## __Not yet implemented:__ Thrifty caching
-By default, Accera caching strategies are *thrifty*, which means that data is physically copied into an allocated cache only if the cached data somehow differs from the original active block. In other words, if the original active block happens to already be contiguous in memory and in the correct memory layout, then Accera skips the caching step and instead just uses the original array. Note that on a GPU, if the cache is supposed to be allocated in a different type of memory than the original array (e.g., the array is in global memory but the cache is supposed to be in shared memory) then a physical copy is created.
+By default, Accera caching strategies are *thrifty* in the sense that the data is physically copied into an allocated cache only if the cached data somehow differs from the original active block. Therefore, if the original active block is already in the correct memory layout and resides contiguously in memory, the Accera skips the caching steps and uses the original array instead. Note that a physical copy is created on a GPU if the cache is supposed to be allocated a different type of memory than the original array (e.g., the array is in global memory, but the cache is supposed to be in shared memory).
 
-For example, say that `A` is a two-dimensional array and its active block at the chosen level is always one of its rows. If `A` is row-major, its rows are already stored contiguously and the data in the active block is identical to the data that would be copied into the cache: both are contiguous and both share the same layout. Since the two are identical, there is no benefit to using the cache over using the original array. The thrifty caching strategy skips the caching step and instead uses the data in the original array. On the other hand, if `A` is column-major, its rows are not stored contiguously. Copying the active row into a contiguous temporary location could be computationally advantageous. In this case, the thrifty caching strategy would create the cache and populate it with data.
+For example, assume that `A` is a two-dimensional array and its active block at the chosen level is always one of its rows. If `A` is row-major, the rows are already stored contiguously. Additionally, data in the active block and data to be copied to cache are identical. This indicates that the data in the active block and the data to be copied into the cache are contiguous and share the same memory layout. Hence, using cache over the original array will be of no use. Therefore, the thrifty caching strategy skips the caching steps and uses the original array instead.
```

Comment:
"Additionally, the data in the active block to be copied to cache are identical: both are contiguous and share the same memory layout. In this case, there is no benefit in using cache over the original array. The thrifty caching strategy will skip the caching steps and use the original array instead."

Reply:
1) data in the active block 
2) data to be copied to cache
These are the two types of data, right? 

In your proposed structure: "Additionally, the data in the active block to be copied to cache are identical: **both**", what does this both represent?

Reply:
oops, I had a typo in my comment (updated now). Thanks for catching it!